src/utils/backend/vaccine.ts

```diff
@@ -0,0 +1,83 @@
+import { BackendDataReturn, DatabaseClient } from "@utils/types/common";
+import { VaccineRecord } from "@utils/types/vaccine";
+
+export async function getVaccineRecordbyPersonId(
+  supabase: DatabaseClient,
+  personId: number
+): Promise<VaccineRecord[]> {
+  const { data, error } = await supabase
+    .from("vaccine_records")
+    .select("*")
+    .eq("person", personId)
+    .order("vaccination_date", { ascending: true });
+
+  if (error) {
+    throw error;
```

Comment:
If you wanna try something new, you should probably ask first. I don’t know anything about this way of reporting errors. If you think this is how we should be moving forward, let’s talk, and then we’ll see how we can migrate.